src/LocalStoreX.ts

```diff
@@ -0,0 +1,243 @@
+import ObjectVersionHelper from './ObjectVersionHelper';
+import IObjectVersionHelper from './type/IObjectVersionHelper';
+import { IStorageItem } from './type/IStorageItem';
+import { isIStorageItem } from './type/typeGuards';
+
+export default class LocalStoreX {
+    /**
+     * LocalStoreX is an instance of a class designed to handle local storage operations.
+     * It provides methods for storing, retrieving, and managing data in the browser's local storage.
+     *
+     * Common use cases include saving user preferences, caching data for offline use,
+     * and persisting application state between sessions.
+     */
+    private static instance: LocalStoreX;
+
+    /**
+     * A default instance of ObjectVersionHelper which implements the IObjectVersionHelper interface.
+     * This instance is commonly used to manage versioning of objects where version control is required.
+     * It provides necessary methods to handle version-related operations seamlessly.
+     */
+    private static defaultVersionHelper: IObjectVersionHelper = new ObjectVersionHelper();
+
+    /**
+     * Constructor for initializing the object with a version helper and performing cleanup of expired items.
+     *
+     * @param {IObjectVersionHelper} objectVersionHelper - An instance used to manage versioning of objects.
+     *
+     * @return {void}
+     */
+    private constructor(private objectVersionHelper: IObjectVersionHelper = LocalStoreX.defaultVersionHelper) {
```

Comment:
An ObjectVersionHelper is useless here, because it can check the structures only on "set", but not on "get". 
For example: you set the data, a version hash is calculated based on object structure, then the developer changes the object structure and makes a release of new app version. So you would try to get data with old structure and the application would crash. This is the purpose why we need versioning, I think generating version number is the responsibility of developer. So we need to extract ObjectVersionHelper and a developer can use it outside of storage.